internal/monitor/driver.go

```diff
@@ -186,3 +186,53 @@ var (
 	deleteLoopBackDevice = utils.DeleteLoopBackDevice
 	unMountPath          = utils.Unmount
 )
+
+//PScaleDriver provides a Driver instance for the PowerScale architecture.
+type PScaleDriver struct {
+}
+
+//GetDriverName returns the driver name string
+func (d *PScaleDriver) GetDriverName() string {
+	return "isilon"
+}
+
+//GetDriverMountDir returns the PowerScale private mount directory.
+func (d *PScaleDriver) GetDriverMountDir(volumeHandle, pvName, podUUID string) string {
+	privateMountDir := os.Getenv("X_CSI_PRIVATE_MOUNT_DIR")
+	if privateMountDir == "" {
+		privateMountDir = "/var/lib/kubelet"
+	}
+	privateMountDir = fmt.Sprintf("%s/pods/%s/volumes/kubernetes.io~csi/%s/mount", privateMountDir, podUUID, pvName)
+	log.Infof("privateMountDir: %s", privateMountDir)
+	return privateMountDir
+}
+
+// GetDriverBlockDev Returns the block device used for a PV by a pod.
+func (d *PScaleDriver) GetDriverBlockDev(volumeHandle, pvName, podUUID string) string {
+	return ""
+}
+
+// GetStagingMountDir Returns the staging directory used by NodeUnstage for a mount device.
+func (d *PScaleDriver) GetStagingMountDir(volumeHandle, pvName string) string {
+	return ""
```

Comment:
Is there not a separate staging directory? What do you expect to be passed as the staging directory to node unstage?

Reply:
PowerScale does not have a separate staging path, we have not implemented NodeStage/NodeUnstage Volume.